### PR TITLE
[Backport to 7.3.x please]LPS-121719 Fix for svg4everybody not actually processing all svgs

### PR DESCRIPTION
**Pull request by wincent**
Based on @ChrisKian's [earlier PR](https://github.com/wincent/liferay-portal/pull/369).

The Liferay code that invokes `svg4everybody` was [updated to include the `validate` option here](https://github.com/liferay-frontend/liferay-portal/pull/104/commits/bee236a18a42b76423b52d7b9c40043ca5c07906).

Due to [this issue in `svg4everybody`](https://github.com/jonathantneal/svg4everybody/issues/148) the code then does not process all svgs, so icons are not always being rendered correctly.

Updating to the `master` branch of `svg4everybody` should fix the problem, but there hasn't been a release in around 3 years, so we "vendor" the code here by copying it along with its license into `frontend-js-svg4everybody-web` and adapting it to our build process.

---

@ChrisKian: I didn't touch the Java changes or even look at them because @petershin already reviewed them [over here](https://github.com/petershin/liferay-portal/pull/1263), but what I did do was squash the commits to avoid the reverts and then changed the way we pull in the JS; instead of using a `dist` (built) version, I added the `lib` (source) and prepended the project's license header. Then I adapted it for our build process and applied our formatting, linting as described in that last commit.

### CODE REVIEW

**Comment by wincent** [comment]
ci:test:sf

**Comment by wincent** [comment]
ci:test:relevant

**Comment by wincent** [comment]
ci:test:bundle

File: modules/apps/frontend-js/frontend-js-svg4everybody-web/src/main/java/com/liferay/frontend/js/svg4everybody/web/internal/servlet/taglib/SVG4EverybodyTopHeadDynamicInclude.java [hunk 1]
```diff
@@ -83,9 +104,12 @@ protected void activate(BundleContext bundleContext) {
 	}
 
 	private static final String[] _JS_FILE_NAMES = {
-		"/svg4everybody/svg4everybody.js"
+		"/index.js"
```
Comment (wincent) [inline modules/apps/frontend-js/frontend-js-svg4everybody-web/src/main/java/com/liferay/frontend/js/svg4everybody/web/internal/servlet/taglib/SVG4EverybodyTopHeadDynamicInclude.java:107]: @jbalsas: Is this simplification I did here kosher? Just to make the URL and the filesystem cleaner... browser still requests `/o/frontend-js-svg4everybody-web/index.js` so there's no ambiguity what this is when you look at it in the network pane.

**Comment by wincent** [comment]
One little thing, I am going to push another commit on top that cleans `svg4everybody` out of the `yarn.lock`, because it is still hanging around in there.

File: modules/apps/frontend-js/frontend-js-svg4everybody-web/src/main/java/com/liferay/frontend/js/svg4everybody/web/internal/servlet/taglib/SVG4EverybodyTopHeadDynamicInclude.java (hunk 1 above)
Comment (jbalsas) [reply modules/apps/frontend-js/frontend-js-svg4everybody-web/src/main/java/com/liferay/frontend/js/svg4everybody/web/internal/servlet/taglib/SVG4EverybodyTopHeadDynamicInclude.java:107]: Looks good enough to me 👍

**Review by jbalsas — APPROVED** [review]
I don't have horrible things to say about this PR

**Comment by wincent** [comment]
ci:test:sf

**Comment by wincent** [comment]
ci:test:relevant

**Comment by wincent** [comment]
ci:test:bundle

**Comment by wincent** [comment]
Just reissued the CI commands because the previous round seemed to have no effect (or maybe their visible effects were canceled when I pushed another commit on top).

**Comment by wincent** [comment]
ci:test:sf

**Comment by wincent** [comment]
ci:test:relevant

**Comment by wincent** [comment]
ci:test:bundle

**Comment by jbalsas** [comment]
@wincent, I gather you manually tested this and we can just forward it?

I edited the title so Brian can automatically backport it to `7.3` and we're off the races :)

Waiting for your confirmation before forwarding (or forward yourself if sure)

**Comment by ChrisKian** [comment]
Hi @jbalsas,

For what it's worth, I just manually tested and confirm if resolves the issue (still).  I confirmed the new index.js file is being used, so my previous fix wasn't cached and being used instead.

CC @wincent

**Comment by wincent** [comment]
Thanks @ChrisKian.

**Comment by wincent** [comment]
ci:forward

**Pull request closed without merging**